### DISCUSSION TITLE
Threads support in Blazor WebAssembly

### DISCUSSION THREAD

**Discussion by fernandreu** [dotnet/aspnetcore#22885] [General] [answered]
I was wondering if you have are any plans to support threads in Blazor WebAssembly, and perhaps some estimated dates for a preview version we could try. I would put this as a comment in #14253 but that issue is now locked.

I think that the scenario might have changed since that issue was created. Now there seems to be a proposal to support threads in WebAssembly, although perhaps the spec is not finished yet. Nevertheless, Chrome seems to have experimental support for it already:
https://developers.google.com/web/updates/2018/10/wasm-threads

Right now, developing a completely standalone Blazor WebAssembly app (i.e. without a server) that involves any CPU-intensive task is not very ideal, given that this blocks the UI thread.

Just for completeness, [this](https://fernando.andreu.info/blazor-fractals/) is what I am using Blazor WebAssembly for (generating fractal images). I tried the `Task.Yield` workaround mentioned in #14253 but removed it because the UI still freezes very noticeably.

**Comment by javiercn** [comment, marked as answer]
@fernandreu thanks for contacting us.

This depends on the maturity of the Browser support + support in the underlying .NET Runtime (mono). It is likely that we move there until all major browsers support threading and the runtime does the work to enable it.
